Add pierce-only detection

alesanmanoweb opened the pull request:
On cut operations that are named "pierce-only" the post processor will detect holes and just pierce a hole in the center.

alesanmanoweb committed:
Add pierce-only detection

On cut operations that are named "pierce-only" the post processor will
detect holes and just pierce a hole in the center.

cmdrcrunchy commented on FireControl.cps:
```diff
@@ -10,8 +10,8 @@
 
-description = "FireControl Plasma v1.6";
-vendor = "Langmuir Systems";
+description = "FireControl Plasma v1.7";
```
dont increment the minor version, instead use 1.6.1

alesanmanoweb replied:
How can this not be categorized as a minor version increment?
https://en.wikipedia.org/wiki/Software_versioning#Semantic_versioning
This PR seems to satisfy the definition of "new, non-breaking feature" used in the article, as it is not just a minor bugfix that an increment to patch level would imply.

cmdrcrunchy commented on FireControl.cps:
```diff
@@ -73,6 +74,7 @@ propertyDefinitions = {
     pierceHeight: {title:"Pierce Height (in)", description:"Height at which to pierce material.", type:"number"},
     cutHeight: {title:"Cut Height (in)", description:"Height at which to cut material.", type:"number"},
     retractHeight: {title:"Retract Height (in)", description:"Height at which to retract above material.", type:"number"},
+    pierceOnlyDetectToggle: {title:"Center pierce only detect", description:"Pierce only the center of holes for cutting operations named 'pierce-only'", type:"boolean"},
```
Awkward title, maybe just put "Pierce Only"

alesanmanoweb replied:
Yeah but this is specifically a hole center pierce, I wanted to convey that... What could I write?

alesanmanoweb committed:
Rounds detection of the location of center of pierce-only holes

Fusion 360 is known to create multiple, partial circular operations (G3) for
a single hole. Sometimes, the center of the sequence of operations can differ
slightly and the pierce-only detection would create several pierce-only
operations for the same hole. By rounding down the location of the center,
we can now detect the same hole and avoid multiple spurious operations.
The detection limit is less than one mm or less than one tenth of an inch.

alesanmanoweb committed:
Add ability to detect multiple pierce-only operations

A complex design might have several "pierce-only" operations and so far it was
impossible for the post-processor to detect more than one. Now we match just
the beginning of the cutr operation name, if it starts with "pierce-only" it
will be treated accordingly.
